src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -38,10 +111,168 @@ private void SetContentDispositionHeader(ActionContext context, FileResult resul
+            var now = DateTimeOffset.UtcNow;
+
+            // 14.25 If-Modified-Since
+            var ifModifiedSince = httpRequestHeaders.IfModifiedSince;
+            if (ifModifiedSince.HasValue && ifModifiedSince <= now)
+            {
+                var modified = ifModifiedSince < lastModified;
+                ifModifiedSinceState = modified ? PreconditionState.ShouldProcess : PreconditionState.NotModified;
+            }
+
+            // 14.28 If-Unmodified-Since
+            var ifUnmodifiedSince = httpRequestHeaders.IfUnmodifiedSince;
+            if (ifUnmodifiedSince.HasValue && ifUnmodifiedSince <= now)
+            {
+                var unmodified = ifUnmodifiedSince >= lastModified;
+                ifUnmodifiedSinceState = unmodified ? PreconditionState.ShouldProcess : PreconditionState.PreconditionFailed;
+            }
+
+            var state = GetMaxPreconditionState(ifMatchState, ifNoneMatchState, ifModifiedSinceState, ifUnmodifiedSinceState);
+            return state;
+        }
+
+        private static PreconditionState GetMaxPreconditionState(params PreconditionState[] states)
+        {
+            var max = PreconditionState.Unspecified;
+            for (var i = 0; i < states.Length; i++)
+            {
+                if (states[i] > max)
+                {
+                    max = states[i];
+                }
+            }
+            return max;
+        }
+
+        private static (RangeItemHeaderValue range, long rangeLength, bool serveBody) SetRangeHeaders(
+            ActionContext context,
+            RequestHeaders httpRequestHeaders,
+            long? fileLength,
+            DateTimeOffset? lastModified = null,
+            EntityTagHeaderValue etag = null)
+        {
+            var response = context.HttpContext.Response;
+            var httpResponseHeaders = response.GetTypedHeaders();
+
+            // Checked for presence of Range header explicitly before calling this method.
+            // Range may be null for parsing errors, multiple ranges and when the file length is missing.
+            var range = fileLength.HasValue ? ParseRange(context, httpRequestHeaders, fileLength.Value, lastModified, etag) : null;
+            var rangeNotSatisfiable = range == null;
+            if (rangeNotSatisfiable)
+            {
+                // 14.16 Content-Range - A server sending a response with status code 416 (Requested range not satisfiable)
+                // SHOULD include a Content-Range field with a byte-range-resp-spec of "*". The instance-length specifies
+                // the current length of the selected resource.  e.g. */length
+                response.StatusCode = StatusCodes.Status416RangeNotSatisfiable;
+                if (fileLength.HasValue)
+                {
+                    httpResponseHeaders.ContentRange = new ContentRangeHeaderValue(fileLength.Value);
+                }
+                return (range: null, rangeLength: fileLength.Value, serveBody: false);
```

Comment:
yes it is ignored essentially since this sets serveBody to false and returns from ExecuteAsync without writing to the response body